jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/stack/Stack.kt

```diff
@@ -206,9 +209,27 @@ private class StackUI(
             }
         })
 
+        actionGroup.addAction(createFilterAction())
+
         return ActionManager.getInstance().createActionToolbar("", actionGroup, false).component
     }
 
+    private fun createFilterAction() =
+        object : ToggleAction(message("cloudformation.stack.filter.show_completed"), null, AllIcons.RunConfigurations.ShowPassed), DumbAware {
+            private val state = AtomicBoolean(true)
```

Comment:
Why do we need the AtomicBoolean? wouldn't it make more sense to do the update synchronously on the ui thread since it's a ui update?

Reply:
the update actually happens in the `Updater` as part of the next 'refresh' call (or one is forced if it's not already inflight).